pkg/common/telemetry/agent/manager.go

```diff
@@ -20,10 +23,7 @@ func StartManagerFetchUpdatesCall(m telemetry.Metrics) *telemetry.CallCounter {
 // Takes registration entry SPIFFE ID
 func IncrRegistrationEntryCreatedCounter(m telemetry.Metrics, id string) {
 	labels := []telemetry.Label{
-		{
-			Name:  telemetry.SPIFFEID,
-			Value: id,
-		},
+		common.GetSanitizedLabel(telemetry.SPIFFEID, id),
```

Comment:
Sanitizing strings here is definitely better than requiring the caller of `IncrRegistrationEntryCreatedCounter` to do it, but I think it's best done as a wrapper around the *WithLabels calls to the metrics implementation. Feels like it would be pretty easy to regress on this as is. Any reason _not_ to apply this sanitizing transform to all labels?

Reply:
`WithLabels` is only being called in one space, by SDS endpoint.

I agree it would be much better to have some kind of wrapping that just ensures all labels always get sanitized, but `WithLabels` seems to not be the place, especially since labels can also be added via `CallCounter` `AddLabel`.
I know that spire allows multiple telemetry plugins active at once, so that must mean there is some delegate object out there that fulfills the Metrics API and calls each of the plugins? Is that `WithLabels` and I just didn't search correctly?

Reply:
hmm seems such an object is `MetricsImpl`, with the desired API being imported from a hashicorp lib object. So don't think this could be done there.
Still not sure how `withLabels` object fits into this, unless you mean that we should propagate its usage beyond the SDS endpoint call?

Reply:
spoke offline for clarity, will fix in next push